aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move

```diff
@@ -380,319 +505,97 @@ module aptos_std::bls12381 {
-            proof_of_possession_from_bytes(x"a6da5f2bc17df70ce664cff3e3a3e09d17162e47e652032b9fedc0c772fd5a533583242cba12095602e422e579c5284b1735009332dbdd23430bbcf61cc506ae37e41ff9a1fc78f0bc0d99b6bc7bf74c8f567dfb59079a035842bdc5fa3a0464"),
-            proof_of_possession_from_bytes(x"b8eef236595e2eab34d3c1abdab65971f5cfa1988c731ef62bd63c9a9ad3dfc9259f4f183bfffbc8375a38ba62e1c41a11173209705996ce889859bcbb3ddd7faa3c4ea3d8778f30a9ff814fdcfea1fb163d745c54dfb4dcc5a8cee092ee0070"),
-            proof_of_possession_from_bytes(x"a03a12fab68ad59d85c15dd1528560eff2c89250070ad0654ba260fda4334da179811d2ecdaca57693f80e9ce977d62011e3b1ee7bb4f7e0eb9b349468dd758f10fc35d54e0d0b8536ca713a77a301944392a5c192b6adf2a79ae2b38912dc98"),
-            proof_of_possession_from_bytes(x"8899b294f3c066e6dfb59bc0843265a1ccd6afc8f0f38a074d45ded8799c39d25ee0376cd6d6153b0d4d2ff8655e578b140254f1287b9e9df4e2aecc5b049d8556a4ab07f574df68e46348fd78e5298b7913377cf5bb3cf4796bfc755902bfdd"),
-        ];
-
-        assert!(std::vector::length(&pks) == std::vector::length(&pops), 1);
-
-        let i = 0;
-        while (i < std::vector::length(&pks)) {
-            let opt_pk = public_key_from_bytes_with_pop(*std::vector::borrow(&pks, i), std::vector::borrow(&pops, i));
-            assert!(option::is_some(&opt_pk), 1);
-
-            i = i + 1;
-        };
-
-        // assert first PK's PoP does not verify against modifed PK' = 0xa0 | PK[1:]
-        let opt_pk = public_key_from_bytes_with_pop(
-            x"a08864c91ae7a9998b3f5ee71f447840864e56d79838e4785ff5126c51480198df3d972e1e0348c6da80d396983e42d7",
-            &proof_of_possession_from_bytes(x"ab42afff92510034bf1232a37a0d31bc8abfc17e7ead9170d2d100f6cf6c75ccdcfedbd31699a112b4464a06fd636f3f190595863677d660b4c5d922268ace421f9e86e3a054946ee34ce29e1f88c1a10f27587cf5ec528d65ba7c0dc4863364"));
-        assert!(option::is_none(&opt_pk), 1);
-
-        // assert first PK's PoP does not verify if modifed as pop' = 0xb0 | pop[1:]
-        let opt_pk = public_key_from_bytes_with_pop(
-            x"808864c91ae7a9998b3f5ee71f447840864e56d79838e4785ff5126c51480198df3d972e1e0348c6da80d396983e42d7",
-            &proof_of_possession_from_bytes(x"bb42afff92510034bf1232a37a0d31bc8abfc17e7ead9170d2d100f6cf6c75ccdcfedbd31699a112b4464a06fd636f3f190595863677d660b4c5d922268ace421f9e86e3a054946ee34ce29e1f88c1a10f27587cf5ec528d65ba7c0dc4863364"));
-        assert!(option::is_none(&opt_pk), 1);
+        let (sk, validated_pk) = generate_keys();
+        let pk_bytes = public_key_with_pop_to_bytes(&validated_pk);
+        let pop = generate_proof_of_possession(&sk);
+        assert!(option::is_some(&public_key_from_bytes_with_pop(pk_bytes, &pop)), 1);
+        assert!(option::is_none(&public_key_from_bytes_with_pop(pk_bytes, &maul_proof_of_possession(&pop))), 1);
```

Comment:
add back this test case too: `// assert first PK's PoP does not verify against modifed PK' = 0xa0 | PK[1:]`